### PR TITLE
build: do enable systemd by default (but don't force it)

### PR DESCRIPTION
**Pull request by jpalus**
contrary to help string (--disable-systemd) systemd was not enabled by default since enable_systemd was set to `no` if no option was passed. do enable it but use `auto` mode by default: use systemd if available but continue if it's missing. enforcing systemd can be ecomplished with --enable-systemd.

### CODE REVIEW

**Review by raveit65 — APPROVED** [review]
LGTM,
build runs fine in CI when systemd is in build root but not forced.

**Review by lukefromdc — APPROVED** [review]
With systemd installed (this is not something I can trivially remove for testing...), this works, and we get systemd units shown in the processes tab w/o having to explicitly set it at build time. Note that this also matches the defaults in meson_options.txt
Approving

**Comment by lukefromdc** [comment]
Is this ready to go or should we wait for another review?

**Pull request merged**
